### PR TITLE
KFSPTS-25319 update version of financials to 8/17/2021

### PR DESCRIPTION
**Pull request by cah292**
There are PRs for this in cu-kfs, nonprod-sql and cu-kfs-upgrades. Please make sure all three are good to go before merging.

This is an updated version of the #1313 PR, which includes some more fixes and cleanup. Here are some of the additional changes I had added on top of Daniela's initial work:

* Adjusted POM dependencies to add in another CXF JAR and to forcibly add the WSS4J JARs that CXF might need, in addition to some other minor cleanup. (Base code removes CXF for this release, but for now we're planning to temporarily keep CXF in the project and to use a future user story to finish removing CXF.)
* Removed the IdentityService overlay that PR #1313 was planning to add. The areas of our code that were using the now-inaccessible getPrincipalsByEntityId() method have been modified to retrieve the same search results through a different service method.
* Added back one of the PurchaseOrderDocument.xml validation entries that PR #1313 was trying to remove, since one of the entries may have been unintentionally removed by that PR.
* Performed some miscellaneous formatting cleanup, to bring things closer to how the equivalent areas of base code have been formatted in the 08/17/2021 release.
  * Note that some of this cleanup involved removing unnecessary parentheses from various boolean condition checks. If you notice any areas where the cleanup may have accidentally changed the expression's logic, please let me know. (When checking this, though, remember that logical AND takes precedence over logical OR, unless parentheses are used to override the order.)

### CODE REVIEW

**Comment by cah292** [comment]
To run the unit tests successfully, we may have to apply the 08/17 upgrade SQL to UTEST (or apply it to UTESTKEW and adjust the Jenkins unit test job accordingly).

**Comment by cah292** [comment]
retest this please

**Comment by cah292** [comment]
retest this please

**Comment by cah292** [comment]
retest this please

**Comment by cah292** [comment]
retest this please

File: pom.xml
Comment (jhulslander) [inline pom.xml:372]: What's this repository?  Should we use our group's repo: nexus.cucloud.net

File: src/main/java/edu/cornell/kfs/kim/service/impl/CuUiDocumentServiceImpl.java
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/kim/service/impl/CuUiDocumentServiceImpl.java:18]: Does KualiCo still want to know when we have to add to overlays to increase visibilities of things?  If so, we should let KualiCo know UiDocumentService's businessObjectService should be protected

**Review by jhulslander — COMMENTED** [review]
I am not done with this review yet, but I found a couple of things to consider

File: pom.xml
Comment (cah292) [reply pom.xml:372]: The plugin repository entries were added to the POM by copying and tweaking all of its regular repository entries. This was one of those existing repository entries in the POM.

If you believe these entries need further cleanup, we could add a follow-up user story to do so unless you think it's urgent to address them now.

File: src/main/java/edu/cornell/kfs/kim/service/impl/CuUiDocumentServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/kim/service/impl/CuUiDocumentServiceImpl.java:18]: Good point, I'll add a note about this in the report-findings-to-KualiCo sub-task.

File: pom.xml
Comment (jhulslander) [reply pom.xml:372]: OK, I see them know.  I think a follow up user story to investigate these entries make sense.  Since these entries were pre-existing and this branch builds, we don't need to adjust this as apart of the kualico patch

**Comment by dp65** [comment]
retest this please

**Comment by dp65** [comment]
please retest this

**Comment by dp65** [comment]
retest this please

**Comment by dp65** [comment]
retest this please

**Comment by dp65** [comment]
retest this please

**Review by jhulslander — COMMENTED** [review]
These changes look good.  I was surprised at some of the "clean up" work that I think makes things hard to read and understand the code.  There were a few places we should make sure the functionals test well, I am not sure if the logic changes keep the functionality in taxt.

**Comment by dp65** [comment]
retest this please

**Comment by dp65** [comment]
retest this please

**Comment by dp65** [comment]
retest this please

File: src/main/java/edu/cornell/kfs/module/purap/document/validation/impl/CuPurchaseOrderAmendmentAccountValidation.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/module/purap/document/validation/impl/CuPurchaseOrderAmendmentAccountValidation.java:83]: I'll add a note for the functionals.

File: src/main/java/edu/cornell/kfs/pdp/service/impl/CuPaymentMaintenanceServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/pdp/service/impl/CuPaymentMaintenanceServiceImpl.java:65]: I'll add a note for the functionals.

**Comment by cah292** [comment]
retest this please

**Comment by cah292** [comment]
It looks like the unit tests passed after the upgrade SQL was applied to the UTEST database. This can be merged once the three 08/17/2021 PRs have all been approved.

File: pom.xml
Comment (cah292) [reply pom.xml:372]: I created KFSPTS-25932 to perform more cleanup work on these repository entries.

**Comment by cah292** [comment]
@jhulslander The unit tests passed after the upgrade SQL was applied to UTEST. The three PRs should be ready to merge once approved.

**Review by jhulslander — APPROVED** [review]
this can be merged now

**Comment by dp65** [comment]
retest this please

**Comment by cah292** [comment]
@dp65 I'm not sure whether the unit tests can still run (or pass) if the branch has already been merged. You may want to test your Jenkins job changes on a new feature branch, or you could potentially try reopening PR #1319 and test with that.

**Pull request merged**
